source/extensions/common/aws/utility.cc

```diff
@@ -234,19 +234,45 @@ Utility::joinCanonicalHeaderNames(const std::map<std::string, std::string>& cano
 }
 
 std::string Utility::getSTSEndpoint(absl::string_view region) {
-  if (region == "cn-northwest-1" || region == "cn-north-1") {
-    return fmt::format("sts.{}.amazonaws.com.cn", region);
+
+  // Handle the scenario where we are using SigV4A and a region set has been provided
+  // SigV4A region sets are comma separated and may include wildcard
+
+  std::string single_region;
+
+  // If we contain a comma or asterisk it looks like a region set
+  if (absl::StrContains(region, ",") || (absl::StrContains(region, "*"))) {
+    // Use the first element from a region set if we have multiple regions specified
```

Comment:
style-nit: Comments start with capital letter and end with '.'.
(Here and in other places)